### PR TITLE
ai: reviewgen controller

### PR DESCRIPTION
**Pull request by acpana**
reviewgen-style chore for controllers for grenefield!

### CODE REVIEW

File: .agents/greenfield-reviewgen-controller.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+*   Verify that `common.CompareProtoMessage` is used to calculate differences between the desired state (`krm` spec mapped to proto) and the actual state (actual proto returned from API).
+*   Verify that the resulting diff paths are used to populate an `UpdateMask` (e.g. `&fieldmaskpb.FieldMask{Paths: sets.List(paths)}`) in the `Update` operation if required by the API.
+
+## 2. Structured Reporting
+*   Verify that `structuredreporting.ReportDiff(ctx, report)` is used in the `Update` method to report the exact diff to the user before submitting the update request to the GCP API.
+
+## 3. Mandatory Status Updates
+*   **Strict Rule:** The controller MUST always update the KRM status at the end of `Create` and `Update` operations, even if there are no fields that need updating in the GCP resource. (e.g., a `setStatus(u, status)` call must be present and invoked, updating `ObservedState`, `ExternalRef`, etc.).
+
+## 4. Wait for LROs (For now)
```
Comment (ldanielmadariaga) [inline .agents/greenfield-reviewgen-controller.md]: I'd add the for now line it may be confusing to the agent. As an alternative you could add a TODO comment to revise this.

File: .agents/greenfield-reviewgen-controller.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+*   **Strict Rule:** For operations that return a Long Running Operation (LRO) from the GCP client, the controller MUST wait for the LRO to finish before proceeding (e.g., calling `op.Wait(ctx)` after `Create` or `Update`).
+
+## 5. Test Coverage (Minimal and Maximal, Edge Cases)
+*   Ensure that the `pkg/test/resourcefixture/testdata/basic/` directory for this resource contains at minimum:
+    *   A minimal test case (`create.yaml`, etc.)
+    *   A maximal test case containing all spec fields.
```
Comment (ldanielmadariaga) [inline .agents/greenfield-reviewgen-controller.md]: Should we add instructions to mention this could be one or more maximal tests and encourage maximal branch coverage? More of a codegen than reviewgen question but...

File: .agents/greenfield-reviewgen-controller.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+*   Check for appropriate edge-case testing if any complex logic exists in the controller (e.g., dependencies on other resources, complex identity mapping).
+
+## 6. General Controller Structure
```
Comment (ldanielmadariaga) [inline .agents/greenfield-reviewgen-controller.md:81]: Link to canonical reference controller implementation would be good here

File: .agents/greenfield-reviewgen-controller.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+## 2. Structured Reporting
+*   Verify that `structuredreporting.ReportDiff(ctx, report)` is used in the `Update` method to report the exact diff to the user before submitting the update request to the GCP API.
+
+## 3. Mandatory Status Updates
+*   **Strict Rule:** The controller MUST always update the KRM status at the end of `Create` and `Update` operations, even if there are no fields that need updating in the GCP resource. (e.g., a `setStatus(u, status)` call must be present and invoked, updating `ObservedState`, `ExternalRef`, etc.).
+
+## 4. Wait for LROs (For now)
+*   **Strict Rule:** For operations that return a Long Running Operation (LRO) from the GCP client, the controller MUST wait for the LRO to finish before proceeding (e.g., calling `op.Wait(ctx)` after `Create` or `Update`).
+
+## 5. Test Coverage (Minimal and Maximal, Edge Cases)
```
Comment (ldanielmadariaga) [inline .agents/greenfield-reviewgen-controller.md:74]: Link to PR or repo file would be good here

**Review by ldanielmadariaga — DISMISSED** [review]

File: .agents/greenfield-reviewgen-controller.md (hunk 1 above)
Comment (acpana) [reply .agents/greenfield-reviewgen-controller.md]: removed!

File: .agents/greenfield-reviewgen-controller.md (hunk 2 above)
Comment (acpana) [reply .agents/greenfield-reviewgen-controller.md]: clarified instructions

File: .agents/greenfield-reviewgen-controller.md (hunk 4 above)
Comment (acpana) [reply .agents/greenfield-reviewgen-controller.md:74]: linked

File: .agents/greenfield-reviewgen-controller.md (hunk 3 above)
Comment (acpana) [reply .agents/greenfield-reviewgen-controller.md:81]: linked w caveat for now

**Review by ldanielmadariaga — APPROVED** [review]
